From 51f1f351262a31d94c10504e4d135526c6a7caeb Mon Sep 17 00:00:00 2001
From: Wang Mingyu <wangmy@cn.fujitsu.com>
Date: Sun, 10 May 2020 21:22:53 +0800
Subject: [PATCH] fix bug of do_compile and do_install

when multiple processes make run in parallel,
because of dependency error will occur.

Signed-off-by: Wang Mingyu <wangmy@cn.fujitsu.com>

[OP: Rebase to 0.9.3]
[OP: Drop dependencies that are already present in the 0.9.3 Makefile]
Signed-off-by: Ovidiu Panait <ovidiu.panait@windriver.com>
---
Upstream-Status: Pending

 Makefile | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/Makefile b/Makefile
index 27b4641f..f195b570 100644
--- a/Makefile
+++ b/Makefile
@@ -79,6 +79,16 @@ libmpathpersist libmpathvalid multipath multipathd: libmultipath
 libmultipath/prioritizers libmultipath/checkers libmultipath/foreign: libmultipath
 mpathpersist multipathd:  libmpathpersist
 
+DEPS_ON_MULTIPATH := \
+	multipath \
+	multipathd \
+	mpathpersist \
+	libmpathpersist
+
+$(DEPS_ON_MULTIPATH:=.install): libmultipath.install
+mpathpersist.install:  libmpathpersist.install
+libdmmp.install libmultipath/foreign.install:  mpathpersist.install
+
 libmultipath/checkers.install \
 	libmultipath/prioritizers.install \
 	libmultipath/foreign.install: libmultipath.install
-- 
2.38.1

